### DISCUSSION TITLE
Experimental Offline Standalone Mac Installer for Stable Diffusion Web UI (unofficial) 20230820

### DISCUSSION THREAD

**Discussion by brkirch** [brkirch/stable-diffusion-webui#33] [Announcements]
This is an offline standalone installer for a modified version of AUTOMATIC1111's Stable Diffusion web UI that includes experimental changes and uses the nightly PyTorch build from 8/14. Note that this is an test build so crashes and bugs should be expected. All dependencies needed to start using Stable Diffusion (other than the model) are included and this does not require Xcode command line tools to be installed.

This release currently only has the installer for Apple silicon Macs, an installer for Intel Macs will be included with 1.6.0 RC. macOS 13.4 or later is highly recommended, older versions of macOS are untested.

<hr /><em>This discussion was created from the release <a href='https://github.com/brkirch/stable-diffusion-webui/releases/tag/20230820_experimental'>Experimental Offline Standalone Mac Installer for Stable Diffusion Web UI (unofficial) 20230820</a>.</em>

**Comment by ericwenger7** [comment]
No trouble installing the 1.6 and SDXL runs well. 

Hi Res fix (text2img) seems to be broken on MAC M2 for all latent mode, the output is scrambled.  

![TKflC](https://github.com/brkirch/stable-diffusion-webui/assets/25514448/a953ca83-ad31-4979-b26d-f3b69a2ffe75)

using a non zero hires step and denoising will be catastrophic in most cases : pure noise is saved at the end. 

Any updates to be able to run ControlNet ?

Is there some documentation about the sdxl-refiner and how to properly use it ?

**Reply by brkirch** [reply]
> Hi Res fix (text2img) seems to be broken on MAC M2 for all latent mode, the output is scrambled.

It's not broken, but according the image metadata the denoising strength was set to 0. Here is what it looks like if you use the default denoising strength of 0.7:
![00002-37596460](https://github.com/brkirch/stable-diffusion-webui/assets/7954378/b59d3671-8bad-40fb-a829-4df9f2c1ffcd)

Also I can see that you aren't using [the LoRA that comes with the SDXL base model](https://huggingface.co/stabilityai/stable-diffusion-xl-base-1.0/resolve/main/sd_xl_offset_example-lora_1.0.safetensors); results will tend to be better with that LoRA so I'd recommend trying it if you haven't already. Here's the image you generated with a denoising strength of 0.7 and the LoRA:
![00003-37596460](https://github.com/brkirch/stable-diffusion-webui/assets/7954378/dd225a17-5196-422c-8da7-e673ef2ba25c)

> Any updates to be able to run ControlNet ?

There is nothing different about this fork that should prevent ControlNet from working correctly. That said, I haven't tested it in a while and I'm currently I'm waiting for [SDXL support](https://github.com/Mikubill/sd-webui-controlnet/issues/1933) before I do anything further with it.

> Is there some documentation about the sdxl-refiner and how to properly use it ?

There isn't really too much to say regarding this - just open the refiner settings, set the refiner model to the model you want to use (for SDXL [`sd_xl_refiner_1.0_0.9vae.safetensors`](https://huggingface.co/stabilityai/stable-diffusion-xl-refiner-1.0/resolve/main/sd_xl_refiner_1.0_0.9vae.safetensors) is recommended), and set `Switch at` to how much of the generation to complete before switching models (e.g. 0.75 completes 75% of the generation before switching to the refiner model). The best results will usually only use the refiner for the last few steps.

**Reply by ericwenger7** [reply]
Thanks for your advices :) 

ControlNet ?
<<
There is nothing different about this fork that should prevent ControlNet from working correctly. That said, I haven't tested it in a while and I'm currently I'm waiting for https://github.com/Mikubill/sd-webui-controlnet/issues/1933 before I do anything further with it. >>
Whenever I try using controlnet I get this console warning :  ControlNet - WARNING - ControlNet does not support SDXL -- disabling
So a script somewhere is disabling it.  Its unclear what script has to be replaced and where to get it... 
(I have ControlNet v1.1.238 installed)

**Comment by ericwenger7** [comment]
I'm finding that the catastrophic noise problem is not linked to hires fix. It does happen randomly and more likely when final size is 1920 and there is no error reported in the console. I'm running on a Mac M2 OS 13
![00050-4024108509](https://github.com/brkirch/stable-diffusion-webui/assets/25514448/fb3e3b0a-1e1f-4700-88d1-9ca077e8c4c8)
On a 25 batch HD generation it happens 25% of the time. Its anoying as it prevents to use batch render for animation or productivity.
![grid-noiseissue](https://github.com/brkirch/stable-diffusion-webui/assets/25514448/75c4d7d4-73a7-4de8-b40f-b2aa7e139f79)

**Reply by brkirch** [reply]
You are using which version of macOS 13? Some older versions (mostly prior to 13.4 IIRC) had various problems with Metal that could cause the problem you describe.

**Comment by ericwenger7** [comment]
> On 23 Aug 2023, at 20:02, brkirch ***@***.***> wrote:
> 
> 
> You are using which version of macOS 13? Some older versions (mostly prior to 13.4 IIRC) had various problems with Metal that could cause the problem you describe.
> 
> 
I thought of that and installed 13.6. 
Still having the problem but it seems less likely :). 
It tend to occur after a couple of correct render when rendering series. I’m often having memory errors so I guess I’m close to the limit with my 32 g.

 The weird thing is sometimes the error shows into the preview but the file is saved correctly, but more often the reverse happen : preview is fine but the file has this beige contour/noise. It usually occurs at the very end of the steps.. I thought it could be related to using fine-tuning but setting it at 1 (never using the refiner) doesn’t change anything.

As a developper I imagine this type of issue can be very hard to trace and understand. 

Anyway I’m getting very nice stuffs with this release :) 
Cheers,
Eric

**Comment by ericwenger7** [comment]
Is there an Offline Standalone Mac Installer  installer for version 1.6 ?

The 1.5 Offline Standalone Mac Installer for Stable Diffusion Web UI was very practical.

Can’t find any installation instruction in the stable-diffusion-webui-1.6.0-RC.zip folder.

Thanks for helping
Eric

**Reply by brkirch** [reply]
https://github.com/brkirch/stable-diffusion-webui/releases/tag/v1.6.0-mac-builds
